fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -539,7 +539,7 @@ private void increaseLimit() {
         }
         final IndexMaintainer maintainer = store.getIndexMaintainer(index);
         final ExecuteProperties.Builder executeProperties = ExecuteProperties.newBuilder()
-                .setIsolationLevel(IsolationLevel.SERIALIZABLE);
+                .setIsolationLevel(IsolationLevel.SNAPSHOT);
```

Comment:
Hm, perhaps this is a sign with our tests if it's not catching a bug here, but I think this needs to validate that the index is idempotent before it sets the isolation level to `SNAPSHOT`. Otherwise, there's not a guarantee that the concurrently written record will be indexed.

We should probably have a comment here explaining why `SNAPSHOT` is used, and then another explaining why we need to add the read conflict range at the place we do.

Reply:
The sum index test did fail, do not lose hope :) 
You did mention before that the SNAPSHOT approach must be limited to idempotent indexes only, but I failed to assimilate (sorry). I'll fix it now.